File: frontend/src/api/sebrae-nacional.ts
import { api } from './client'
import type { NacionalDashboard } from '../types/dashboard'

export async function buscarDashboard(
  periodoInicio?: string,
  periodoFim?: string,
): Promise<NacionalDashboard> {
  const params: Record<string, string> = {}
  if (periodoInicio) params.periodo_inicio = periodoInicio
  if (periodoFim) params.periodo_fim = periodoFim
  return api.get<NacionalDashboard>('/api/v1/sebrae-nacional/dashboard', params)
}

export async function buscarMapa() {
  return api.get('/api/v1/sebrae-nacional/mapa')
}

export async function buscarDesertos() {
  return api.get('/api/v1/sebrae-nacional/desertos')
}

export async function buscarTendencias(periodoMeses = 12) {
  return api.get('/api/v1/sebrae-nacional/tendencias', {
    periodo_meses: String(periodoMeses),
  })
}

File: backend/app/routers/sebrae_nacional.py
"""Router do painel SEBRAE Nacional."""

from datetime import date
from typing import Optional

from fastapi import APIRouter, Depends, Query
from sqlalchemy.ext.asyncio import AsyncSession

from app.database import get_db
from app.services import dashboard_nacional

router = APIRouter()


@router.get("/dashboard")
async def dashboard(
    periodo_inicio: Optional[date] = Query(None, description="Inicio do periodo (YYYY-MM-DD)"),
    periodo_fim: Optional[date] = Query(None, description="Fim do periodo (YYYY-MM-DD)"),
    db: AsyncSession = Depends(get_db),
):
    return await dashboard_nacional.montar_dashboard(db, periodo_inicio, periodo_fim)


@router.get("/mapa")
async def mapa(
    db: AsyncSession = Depends(get_db),
):
    return await dashboard_nacional.dados_mapa(db)


@router.get("/desertos")
async def desertos(
    db: AsyncSession = Depends(get_db),
):
    return await dashboard_nacional.desertos_fornecimento(db)


@router.get("/tendencias")
async def tendencias(
    periodo_meses: int = Query(12, ge=1, le=60, description="Quantidade de meses para a serie temporal"),
    db: AsyncSession = Depends(get_db),
):
    return await dashboard_nacional.tendencias(db, periodo_meses)
